tests/ignite/metrics/test_psnr.py

```diff
@@ -0,0 +1,236 @@
+import os
+
+import numpy as np
+import pytest
+import torch
+from skimage.metrics import peak_signal_noise_ratio as ski_psnr
+
+import ignite.distributed as idist
+from ignite.exceptions import NotComputableError
+from ignite.metrics import PSNR
+from ignite.utils import manual_seed
+
+
+def test_zero_div():
+    psnr = PSNR()
+    with pytest.raises(NotComputableError, match="PSNR must have at least one example before it can be computed"):
+        psnr.compute()
+
+
+def test_invalid_psnr():
+    y_pred = torch.rand(1, 3, 8, 8)
+    y = torch.rand(1, 3, 8, 8)
+
+    with pytest.raises(TypeError, match=r"Expected y_pred and y to have the same data type."):
+        psnr = PSNR()
+        psnr.update((y_pred, y.double()))
+
+    with pytest.raises(ValueError, match=r"Expected y_pred and y to have the same shape."):
+        psnr = PSNR()
+        psnr.update((y_pred, y.squeeze(dim=0)))
+
+    with pytest.raises(ValueError, match=r"y has intensity values outside the range expected"):
+        psnr = PSNR()
+        psnr.update((y_pred, y))
+        # to catch ValueError for this batch
+        psnr.update((y_pred, y + 1.0))
+
+
+def _test_psnr(y_pred, y, data_range, device):
+    psnr = PSNR(data_range=data_range, device=device)
+    psnr.update((y_pred, y))
+    psnr_compute = psnr.compute()
+
+    np_y_pred = y_pred.cpu().numpy()
+    np_y = y.cpu().numpy()
+
+    assert isinstance(psnr_compute, torch.Tensor)
+    assert psnr_compute.dtype == torch.float64
+    assert psnr_compute.device == torch.device(device)
+    assert np.allclose(psnr_compute.numpy(), ski_psnr(np_y, np_y_pred, data_range=data_range))
```

Comment:
Skimage can also compute on PSNR on batch of images ? (Haven't checked their doc yet)

Reply:
Yes, it does

Reply:
No, actually, it considers batch of images as N-dim image and compute psnr on batch images considering it as a single image and we would like to have average of B psnr's.